### PR TITLE
feat: Implement CLIENT PAUSE

### PR DESCRIPTION
**Pull request by siyopao**
Part of #174.

Hello :wave: My first contribution! Please let me know if it can be improved. Thanks!

### CODE REVIEW

**Comment by siyopao** [comment]
Oh, the test fails. I think PAUSE with the mode parameter only works on Redis >= 6.2.0 Also UNPAUSE. So I could remove the mode and UNPAUSE or... any suggestions?

File: command.ts [hunk 1]
```diff
@@ -876,13 +876,27 @@ XRANGE somestream - +
...
    * You can get the name of the current connection using `client_getname()`.
    * @see https://redis.io/commands/client-setname
    */
   client_setname(connectionName: string): Promise<Status>;
 
+  /**
+   * Used to resume command processing for all clients that were paused by `client_pause()`.
+   * @see https://redis.io/commands/client-unpause
+   */
+  client_unpause(): Promise<Status>;
```
Comment (uki00a) [inline command.ts:898]: Please remove `client_unpause` method!

**Review by uki00a — CHANGES_REQUESTED** [review]
@siyopao Thank you for the contribution! 

> Oh, the test fails. I think PAUSE with the mode parameter only works on Redis >= 6.2.0. Also UNPAUSE. So I could remove the mode and UNPAUSE or... any suggestions?

OK, let's leave those for now. We'll try to support those in a future version. :wink:

File: command.ts [hunk 2]
```diff
@@ -876,13 +876,27 @@ XRANGE somestream - +
    */
   client_id(): Promise<Integer>;
 
+  /**
+   * Suspend all the Redis clients for the specified amount of time (in milliseconds). Currently supports two modes:
+   * ALL: This is the default mode. All client commands are blocked.
+   * WRITE: Clients are only blocked if they attempt to execute a write command.
+   * @see https://redis.io/commands/client-pause
+   */
+  client_pause(timeout: number, mode?: "WRITE" | "ALL"): Promise<Status>;
```
[the remark this answers is not in the document]
Comment (siyopao) [reply command.ts]: Updated! Thanks.

File: command.ts [hunk 3]
```diff
@@ -876,13 +876,27 @@ XRANGE somestream - +
    */
   client_id(): Promise<Integer>;
 
+  /**
```
[the remark this answers is not in the document]
Comment (siyopao) [reply command.ts:879]: Fixed! Thanks.

File: redis.ts [hunk 4]
```diff
@@ -291,10 +291,21 @@ export class RedisImpl implements Redis {
     return this.execIntegerReply("CLIENT", "ID");
   }
 
+  client_pause(timeout: number, mode?: "WRITE" | "ALL") {
```
[the remark this answers is not in the document]
Comment (siyopao) [reply redis.ts]: Removed. Thanks!

File: command.ts (hunk 1 above)
Comment (siyopao) [reply command.ts]: Removed for now. Thanks!

**Review by uki00a — APPROVED** [review]
LGTM, thank you for your contribution! :+1:

**Comment by siyopao** [comment]
Thanks for the great experience! I hope I can contribute more in the future.

**Pull request merged**